### PR TITLE
core: ensure coinbase account is created during block processing

### PR DESCRIPTION
**Pull request by Gustav-Simonsson**

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Tue Sep 1 15:18:53 UTC 2015_

File: core/block_processor.go [hunk 1]
```diff
@@ -81,8 +81,10 @@ func (sm *BlockProcessor) TransitionState(statedb *state.StateDB, parent, block
 	return receipts, nil
 }
 
-func (self *BlockProcessor) ApplyTransaction(coinbase *state.StateObject, statedb *state.StateDB, header *types.Header, tx *types.Transaction, usedGas *big.Int, transientProcess bool) (*types.Receipt, *big.Int, error) {
-	cb := statedb.GetStateObject(coinbase.Address())
+func (self *BlockProcessor) ApplyTransaction(statedb *state.StateDB, header *types.Header, tx *types.Transaction, usedGas *big.Int, transientProcess bool) (*types.Receipt, *big.Int, error) {
+	cb := statedb.GetOrNewStateObject(header.Coinbase)
+	cb.SetGasLimit(header.GasLimit)
```
Comment (karalabe) [inline core/block_processor.go]: Why do you need to set a gas limit here?

Comment (Gustav-Simonsson) [inline core/block_processor.go]: It's done at beginning of block processing before first tx is processed: https://github.com/ethereum/go-ethereum/blob/fd512fa12c59657d9e47cc3411e6e24bd1af89cb/core/block_processor.go#L73

It's used in further checks - though now that I think of it, we have to ensure that we do not reset how much gas has been used so far in the block, as the block's gasLimit applies for the full set of txs.

Comment (subtly) [inline core/block_processor.go]: @karalabe Can you check PMs on skype?

Comment (Gustav-Simonsson) [inline core/block_processor.go]: Yepp we got to make sure gasUsed (gasPool) is correctly tracked, closing this PR and let's re-open a new to develop branch once we figure out the correct tracking.

**Pull request closed without merging**